Build ADI LDR format executable for U-Boot Proper

pamolloy opened the pull request:
I believe this is missing in upstream since the focus was on Falcon mode, but the default should be to support U-Boot proper as with all other boards and then disable `u-boot.ldr` when Falcon mode is enabled.

pamolloy commented:
Hi @gmalysa, when I was building mainline (including with your series applied) I noticed `u-boot.ldr` for U-Boot Proper wasn't getting built. Hopefully I didn't miss something like enabling a Kconfig option. If not then I'll submit this upstream.

gmalysa commented:
Historically we didn't do any standalone u-boot, so everything was expected to go through yocto, where proper would be packaged into an ldr as the stage 2 boot package, potentially with additional binaries depending on security options in the same file (or, in falcon, i believe the plan was to pack the initrd + kernel into the ldr instead). I can't remember why we originally ended up with the split (spl is packaged, proper is not) instead of doing both or neither, but it's probably just that stage 1 never needs to be repackaged and stage 2 *sometimes* does, and each file can only be modified by one package, so we needed proper's .elf but never the .ldr, as a new stage 2 ldr had to come from adsp-boot.

There's no extra Kconfig; I had considered adding the u-boot.ldr step as well previously, but I think it caused build failures in CI so I had postponed it until the ldr tool was added to CI (part of the last series being mainlined now).

pamolloy committed:
mach-sc5xx: generate U-Boot proper in ADI ldr format

Generating an ldr boot stream containing U-Boot Proper was never added
to U-Boot because it is done by the ADI Yocto layer. Add it to U-Boot to
support projects that do not use that layer.

Signed-off-by: Philip Molloy <philip.molloy@analog.com>
(cherry picked from commit 55014ce40e29fc4a8d5ec146b02ea0c660b03087)

pamolloy commented:
https://lore.kernel.org/u-boot/20250814135456.2496908-1-philip@philipmolloy.com/T/#u
https://lore.kernel.org/u-boot/20250814152733.2514599-1-philip@philipmolloy.com/T/#u

pamolloy commented:
```
Applied to u-boot/next, thanks!

[1/1] mach-sc5xx: generate U-Boot proper in ADI ldr format
      commit: 55014ce40e29fc4a8d5ec146b02ea0c660b03087
```

https://lore.kernel.org/u-boot/175581002748.928164.16873234591986258474.b4-ty@konsulko.com/T/#m55b003629baab1667a44cfa32fbf7ae1a9cd1451

pamolloy commented:
This is ready to be merged. I've cherry-picked the mainline commit onto this branch and force pushed over the original commit.
